homework/Repository.js

```diff
@@ -12,9 +12,29 @@ class Repository {
    * Render the repository info to the DOM.
    * @param {HTMLElement} container The container element in which to render the repository.
    */
-  render(container) {
+  render(leftDiv) {
     // TODO: replace the next line with your code.
-    Util.createAndAppend('pre', container, JSON.stringify(this.repository, null, 2));
+    const table = Util.createAndAppend('table', leftDiv, { class: 'table' });
+    const tBody = Util.createAndAppend('tbody', table);
+    const repositoryRow = Util.createAndAppend('tr', tBody, { class: 'row' });
+    Util.createAndAppend('td', repositoryRow, { text: 'Repository :', class: 'label' });
+    const repositoryRowSecondTd = Util.createAndAppend('td', repositoryRow);
+    Util.createAndAppend('a', repositoryRowSecondTd, {
+      href: this.repository.html_url,
+      target: '_blank',
+      text: this.repository.name,
+    });
+
+    function appendRow(parent, text1, text2) {
```

Comment:
Avoid embedded names functions (other than unnamed callbacks and event handlers). They cause the function body of the enclosing to become larger (and therefore more cumbersome to oversee). In this case, since this function does not need anything from the object instance, you could make it a static method, like this:

```js
render(leftDiv) {
  ...
   if (this.repository.description !== null) {
    Repository.appendRow(tBody, 'Description :', this.repository.description);
  }
  Repository.appendRow(tBody, 'Forks :', this.repository.forks);
  Repository.appendRow(tBody, 'Updated :', new Date(this.repository.updated_at).toLocaleString());
}

static appendRow(parent, text1, text2) {
  const tr = Util.createAndAppend('tr', parent, { class: 'row' });
  Util.createAndAppend('td', tr, { text: text1, class: 'label' });
  Util.createAndAppend('td', tr, { text: text2 });
}
```

Reply:
yes, after learning class static function, using it will be better.

Reply:
Thanks for your good wishes. And Thank you for your all efforts.